code/Workable.php

```diff
@@ -1,113 +1,157 @@
-
-	/**
-	 * Reference to the RestfulService dependency
-	 * @var RestfulService
-	 */
-	protected $restulService;
-
-	/**
-	 * Constructor, inject the restful service dependency
-	 * @param RestfulService $restfulService
-	 */
-	public function __construct($restfulService) {
-		$this->restfulService = $restfulService;
-
-		parent::__construct();
-	}
-
-	/**
-	 * Gets all the jobs from the Workable API
-	 * @param  array  $params Array of params, e.g. ['state' => 'published']
-	 * @return ArrayList
-	 */
-	public function getJobs($params = []) {
-		$list = ArrayList::create();
-		$response = $this->callRestfulService('jobs', $params);
-
-		if($response && isset($response['jobs']) && is_array($response['jobs'])) {			
-			foreach($response['jobs'] as $record) {
-				$list->push(Workable_Result::create($record));
-			}
-		}
-
-		return $list;
-	}
-
-	/**
-	 * Wrapper method to configure the RestfulService, make the call, and handle errors
-	 * @param  string $url    
-	 * @param  array  $params 
-	 * @param  string $method 
-	 * @return array         JSON
-	 */
-	protected function callRestfulService($url, $params = [], $method = 'GET') {
-		$this->restfulService->setQueryString($params);
-		$response = $this->restfulService->request($url, $method, $params);
-		
-		if(!$response) {
-			SS_Log::log('No response from workable API endpoint ' . $url, SS_Log::WARN);
-			
-			return false;				
-		}
-		else if($response->getStatusCode() !== 200) {
-			SS_Log::log("Received non-200 status code {$response->getStatusCode()} from workable API", SS_Log::WARN);
-
-			return false;
-		}
-
-		return Convert::json2array($response->getBody());
-	}
-
+namespace SilverStripe\Workable;
+
+use Monolog\Logger;
+use RuntimeException;
+use Psr\Log\LoggerInterface;
+use SilverStripe\Core\Convert;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\Core\Flushable;
+use SilverStripe\View\ArrayData;
+use SilverStripe\Core\Extensible;
+use Psr\SimpleCache\CacheInterface;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+
+class Workable implements Flushable
+{
+    use Extensible;
+    use Injectable;
+    use Configurable;
+
+    /**
+     * Reference to the RestfulService dependency
+     * @var RestfulService
+     */
+    protected $restulService;
+
+    /**
+     * Constructor, inject the restful service dependency
+     * @param RestfulService $restfulService
+     */
+    public function __construct($restfulService)
+    {
+        $this->restfulService = $restfulService;
+    }
+
+    /**
+     * Gets all the jobs from the Workable API
+     * @param  array  $params Array of params, e.g. ['state' => 'published'].
+     *                        see https://workable.readme.io/docs/jobs for full list of query params
+     * @return ArrayList
+     */
+    public function getJobs($params = [])
+    {
+        $cache = Injector::inst()->get(CacheInterface::class . '.workable');
```

Comment:
The way this is written, `CacheInterface` is an implicit dependency of this class. Let's add it as a constructor arg or at least an injected property, and chuck it into the config.